samples/matter/common/src/binding/binding_handler.cpp

```diff
@@ -15,149 +15,176 @@ using namespace chip::app;
+{
+	BindingTable &bindingTable = BindingTable::GetInstance();
+
+	LOG_INF("Binding Table size: [%d]:", bindingTable.Size());
+	uint8_t i = 0;
+	for (auto &entry : bindingTable) {
+		switch (entry.type) {
+		case EMBER_UNICAST_BINDING:
+			LOG_INF("[%d] UNICAST:", i++);
+			LOG_INF("\t\t+ Fabric: %d\n \
             \t+ LocalEndpoint %d \n \
             \t+ ClusterId %d \n \
             \t+ RemoteEndpointId %d \n \
             \t+ NodeId %d",
-					(int)entry.fabricIndex, (int)entry.local, (int)entry.clusterId.Value(),
-					(int)entry.remote, (int)entry.nodeId);
-				break;
-			case EMBER_MULTICAST_BINDING:
-				LOG_INF("[%d] GROUP:", i++);
-				LOG_INF("\t\t+ Fabric: %d\n \
+				(int)entry.fabricIndex, (int)entry.local, (int)entry.clusterId.Value(),
+				(int)entry.remote, (int)entry.nodeId);
+			break;
+		case EMBER_MULTICAST_BINDING:
+			LOG_INF("[%d] GROUP:", i++);
+			LOG_INF("\t\t+ Fabric: %d\n \
             \t+ LocalEndpoint %d \n \
             \t+ RemoteEndpointId %d \n \
             \t+ GroupId %d",
-					(int)entry.fabricIndex, (int)entry.local, (int)entry.remote,
-					(int)entry.groupId);
-				break;
-			case EMBER_UNUSED_BINDING:
-				LOG_INF("[%d] UNUSED", i++);
-				break;
-			default:
-				break;
-			}
+				(int)entry.fabricIndex, (int)entry.local, (int)entry.remote, (int)entry.groupId);
+			break;
+		case EMBER_UNUSED_BINDING:
+			LOG_INF("[%d] UNUSED", i++);
+			break;
+		default:
+			break;
 		}
 	}
+}
+
+void BindingHandler::DeviceWorkerHandler(intptr_t context)
+{
+	VerifyOrDie(context != 0);
+	BindingData *data = reinterpret_cast<BindingData *>(context);
+
+	if (BindingTable::GetInstance().Size() != 0) {
+		LOG_INF("Notify Bounded Cluster | endpoint: %d cluster: %d", data->EndpointId, data->ClusterId);
+		BindingManager::GetInstance().NotifyBoundClusterChanged(data->EndpointId, data->ClusterId,
+									static_cast<void *>(data));
+	} else {
+		LOG_INF("NO DEVICE BOUND");
+	}
+}
 
-	void BindingHandler::DeviceWorkerHandler(intptr_t context)
-	{
-		VerifyOrDie(context != 0);
-		BindingData *data = reinterpret_cast<BindingData *>(context);
-
-		if (BindingTable::GetInstance().Size() != 0) {
-			LOG_INF("Notify Bounded Cluster | endpoint: %d cluster: %d", data->EndpointId, data->ClusterId);
-			BindingManager::GetInstance().NotifyBoundClusterChanged(data->EndpointId, data->ClusterId,
-										static_cast<void *>(data));
-		} else {
-			LOG_INF("NO DEVICE BOUND");
+CHIP_ERROR BindingHandler::RegisterBindingEventHandler(ReleaseCallback bindingReleaseCallback)
+{
+	ReturnErrorOnFailure(Nrf::Matter::RegisterEventHandler(BindingMatterEventHandler, 0));
+
+	mReleaseCallbackPointer = bindingReleaseCallback;
+
+	return CHIP_NO_ERROR;
+}
+
+void BindingHandler::BindingMatterEventHandler(const DeviceLayer::ChipDeviceEvent *event, intptr_t /* unused */)
+{
+	switch (event->Type) {
+	case DeviceLayer::DeviceEventType::kBindingsChangedViaCluster:
+		if (BindingTable::GetInstance().Size() == 0) {
```

Comment:
This won't really work if there are multiple bindings. Not sure if this can be made generically without modifying the Matter core as the event carries no information about the removed binding, so only the application can react to `kBindingsChangedViaCluster` properly by checking if the concerned binding is still present in the table.

Reply:
With kBindingsChangedViaCluster event, only fabricIndex value is given. Is it enough to identify which connection changed?

Reply:
Unfortunately not. kBindingsChangedViaCluster is generated once per entire binding table update and there might be multiple entries changed in one update. I guess you could replace `ReleaseCallback` with `UpdateCallback` and have the application determine in `UpdateCallback` if an entry for your thermostat exists.

Reply:
Yeah, that's good idea. Thanks!